provider/pkg/arm2pulumi/internal/test/quickstart_coverage_test.go

```
@@ -71,23 +72,26 @@ func TestQuickstartTemplateCoverage(t *testing.T) {
 					Template: template,
 				})
 			}
-			// Bail on this test if conversion failed.
-			require.NoError(t, err)
 
-			if len(diags) == 0 {
-				// Success is represented by no diagnostic information apart from template name.
-				diagList = append(diagList, Diag{Template: template})
-			}
-			for _, v := range diags {
-				for _, d := range v {
-					diagList = append(diagList, Diag{Template: template, Diagnostic: d})
+			if err == nil {
+				if len(diags) == 0 {
+					// Success is represented by no diagnostic information apart from template name.
+					diagList = append(diagList, Diag{Template: template})
 				}
+				for _, v := range diags {
+					for _, d := range v {
+						diagList = append(diagList, Diag{Template: template, Diagnostic: d})
+					}
+				}
+				if *retainConverted {
+					rendered, _, err := renderer.RenderPrograms(body, []string{"nodejs"})
+					require.NoError(t, err)
+					require.NoError(t, ioutil.WriteFile(filepath.Join(*testOutputDir, template+".ts"), []byte(rendered["nodejs"]), 0600))
+				}
+			} else {
+				t.Log(err) // Silently log the error if this conversion failed
```

I think we might want to record this as a fatal error instead of swallowing the error? Just create a new diagnostic with fatal and record something relevant to track the affected template.

---

I thought lines 66-74 was still doing that, but I could misunderstand? If we record this as a fatal error do you mean doing something like `t.Error(err)` -- that seems to cause the overall test to fail. Or are you referring to something like `diagList` as an internal way to store the data?

---

Ah so it is! You can just get rid of the else block IMO. We have what we want.